common/utils.py

```diff
@@ -350,46 +394,56 @@ def get_item_topics(param):
              topic hierarchy, its corresponding item id,
              and a placeholder number of rights.
     """
+    catch_error = get_error(param)
+    if catch_error[0]:
+        return catch_error[1]
     inp = inp = update_input(param)
-    if inp == get_keyword_value("no_students"):
-        return get_keyword_value("no_students")
     topics = inp.get(get_keyword_value("item_topics"))
     
     if not topics:
         return get_keyword_value("no_topics")
 
+    item_ids = []
+    for i in topics:
+        item = i.get(get_keyword_value("item_id"))
+        if item:
+            if item not in item_ids:
+                item_ids.append(item)
+
     tree_dict = {}
-    index = 0
+    index = 1
     for i in topics:
-        index += 1
-        item = i.get(get_keyword_value("item_id"), index)
+        item = i.get(get_keyword_value("item_id"))
+        if not item:
+            while index in item_ids:
+                index += 1
+            item = index
+            item_ids.append(index)
         tree_dict[item] = []
-        tags = i.get(get_keyword_value("tags"))
-        if tags:
-            for k in tags:
-                if k.get(get_keyword_value("scored"), "Y") != "Y":
-                    continue
-                curr_tree = k.get(get_keyword_value("topic_tree"), 
-                            inp[get_keyword_value("exam")][get_keyword_value("name")])
-                curr_levels = k.get(get_keyword_value("topic_branch_hierarchy"))
-                curr_topic = k.get(get_keyword_value("topic_tagged"), get_keyword_value("unknown"))
-
-                level_list = []
-                if curr_levels:
-                    for i in curr_levels:
-                        level_list.append(int(i))
-                    level_list.sort(reverse=True)
-
-                hier_level = {}
-                hier_level[0] = curr_tree
-                if curr_levels:
-                    for i in range(1, level_list[0]+2):
-                        hier_level[i] = curr_levels.get(str(i-1), "Unkown")
-                    hier_level[level_list[0]+2] = curr_topic
-                else:
-                    hier_level[1] = curr_topic
-
-                tree_dict[item].append(hier_level)
```

Comment:
This is the new code. Also on line 387 can you change `"Unknown"` to `get_keyword_value("unknown")`

Reply:
it is included on line 423 onward. I will fix 

> Unknown

 thing

Reply:
The code from lines 423 are incorrect. I highlighted the differences here:
![image](https://user-images.githubusercontent.com/61991709/80549482-fc4ed500-8982-11ea-9e22-f1a11609d12f.png)

The things I put in red are what we need to keep.

Reply:
All tests are passing. You can do that change and do a PR.

Reply:
ok